alica_test_utility/src/TestContext.cpp

```diff
@@ -43,132 +48,54 @@ bool TestContext::makeBehaviourEventDriven(int64_t behaviourID)
-        std::unique_ptr<BehaviourTrigger> trigger = std::make_unique<BehaviourTrigger>();
-        behaviour->setTrigger(trigger.get());
-        return std::move(trigger);
+    // register the trigger in the TestContext trigger-map
+    auto behaviourConfKey = std::make_pair(behaviourID, configurationID);
+    auto behaviourTriggerEntry = _behaviourTriggers.find(behaviourConfKey);
+    if (behaviourTriggerEntry == _behaviourTriggers.end()) {
+        _behaviourTriggers[behaviourConfKey] = behaviourTrigger;
     } else {
-        return nullptr;
+        std::cerr << "[TestContext] Trigger for behaviourID " << behaviourID << " and configurationID " << configurationID << " already exists!" << std::endl;
     }
-}
-
-void TestContext::handleAgentAnnouncement(alica::AgentAnnouncement agentAnnouncement)
-{
-    _engine->editTeamManager().handleAgentAnnouncement(agentAnnouncement);
-}
-
-void TestContext::setTeamTimeout(AlicaTime timeout)
-{
-    _engine->editTeamManager().setTeamTimeout(timeout);
-}
-
-void TestContext::tickTeamManager()
-{
-    _engine->editTeamManager().tick();
-}
-
-void TestContext::tickTeamObserver(alica::RunningPlan* root)
-{
-    _engine->editTeamObserver().tick(root);
-}
-
-void TestContext::tickRoleAssignment()
-{
-    _engine->editRoleAssignment().tick();
-}
-
-alica::PlanSelector* TestContext::getPlanSelector()
-{
-    return _engine->getPlanBase().getPlanSelector();
-}
-
-alica::RunningPlan* TestContext::makeRunningPlan(const alica::Plan* plan, const alica::Configuration* configuration)
-{
-    return _engine->editPlanBase().makeRunningPlan(plan, configuration);
-}
-
-VariableSyncModule& TestContext::editResultStore() {
-    return _engine->editResultStore();
-}
-
-//////////////////////////////////////////////////////
-// Getter for tests ... everything returned is const//
-//////////////////////////////////////////////////////
-
-const RunningPlan* TestContext::getDeepestNode()
-{
-    return _engine->getPlanBase().getDeepestNode();
-}
-
-const RunningPlan* TestContext::getRootNode()
-{
-    return _engine->getPlanBase().getRootNode();
-}
 
-const BlackBoard& TestContext::getBlackBoard()
-{
-    return _engine->getBlackBoard();
+    // return the registered trigger
+    return behaviourTrigger;
 }
 
-const BehaviourPool& TestContext::getBehaviourPool()
+std::shared_ptr<BasicBehaviour> TestContext::getBasicBehaviour(int64_t behaviourID, int64_t configurationID)
 {
-    return _engine->getBehaviourPool();
+    assert(_initCalled == true);
+    return Util::getBasicBehaviour(_engine.get(), behaviourID, configurationID);
 }
 
 int TestContext::getTeamSize()
 {
-    return _engine->getTeamManager().getTeamSize();
-}
-
-const DomainVariable* TestContext::getDomainVariable(essentials::IdentifierConstPtr agentID, std::string sort)
-{
-    return _engine->getTeamManager().getDomainVariable(agentID, sort);
+    return Util::getTeamSize(_engine.get());
 }
 
 const alica::Agent* TestContext::getLocalAgent()
 {
```

Comment:
Don't need to expose the internal Agent class

Reply:
removed